feature_extractor/item_helper.py

```diff
@@ -0,0 +1,443 @@
+from typing import Dict, Any
+
+import numpy as np
+
+import nmmo
+from nmmo.core.tile import TileState
+from nmmo.entity.entity import EntityState
+from nmmo.lib import material
+from nmmo.systems import item as Item
+from nmmo.systems.item import ItemState
+
+from model.model import ModelArchitecture
+
+from feature_extractor.entity_helper import EntityHelper
+
+EntityAttr = EntityState.State.attr_name_to_col
+ItemAttr = ItemState.State.attr_name_to_col
+TileAttr = TileState.State.attr_name_to_col
+
+ITEM_TO_PROF_LEVEL = {
+  Item.Hat.ITEM_TYPE_ID: "level",
+  Item.Top.ITEM_TYPE_ID: "level",
+  Item.Bottom.ITEM_TYPE_ID: "level",
+  Item.Sword.ITEM_TYPE_ID: "melee_level",
+  Item.Bow.ITEM_TYPE_ID: "range_level",
+  Item.Wand.ITEM_TYPE_ID: "mage_level",
+  Item.Scrap.ITEM_TYPE_ID: "melee_level",
+  Item.Shaving.ITEM_TYPE_ID: "range_level",
+  Item.Shard.ITEM_TYPE_ID: "mage_level",
+  Item.Rod.ITEM_TYPE_ID: "fishing_level",
+  Item.Gloves.ITEM_TYPE_ID: "herbalism_level",
+  Item.Pickaxe.ITEM_TYPE_ID: "prospecting_level",
+  Item.Chisel.ITEM_TYPE_ID: "carving_level",
+  Item.Arcane.ITEM_TYPE_ID: "alchemy_level",
+  Item.Ration.ITEM_TYPE_ID: "level",
+  Item.Poultice.ITEM_TYPE_ID: "level",
+}
+
+ARMORS = {
+  Item.Hat.ITEM_TYPE_ID,
+  Item.Top.ITEM_TYPE_ID,
+  Item.Bottom.ITEM_TYPE_ID,
+}
+
+WEAPONS = {
+  Item.Sword.ITEM_TYPE_ID,
+  Item.Wand.ITEM_TYPE_ID,
+  Item.Bow.ITEM_TYPE_ID,
+}
+
+TOOLS = {
+  Item.Rod.ITEM_TYPE_ID,
+  Item.Gloves.ITEM_TYPE_ID,
+  Item.Pickaxe.ITEM_TYPE_ID,
+  Item.Chisel.ITEM_TYPE_ID,
+  Item.Arcane.ITEM_TYPE_ID,
+}
+
+AMMOS = {
+  Item.Scrap.ITEM_TYPE_ID,
+  Item.Shaving.ITEM_TYPE_ID,
+  Item.Shard.ITEM_TYPE_ID,
+}
+
+CONSUMABLES = {
+  Item.Ration.ITEM_TYPE_ID,
+  Item.Poultice.ITEM_TYPE_ID,
+}
+
+ATK_TO_WEAPON = {
+  'Melee': Item.Sword.ITEM_TYPE_ID,
+  'Range': Item.Bow.ITEM_TYPE_ID,
+  'Mage': Item.Wand.ITEM_TYPE_ID
+}
+
+ATK_TO_TOOL = {
+  'Melee': Item.Pickaxe.ITEM_TYPE_ID,
+  'Range': Item.Chisel.ITEM_TYPE_ID,
+  'Mage': Item.Arcane.ITEM_TYPE_ID
+}
+
+ATK_TO_TILE = {
+  'Melee': material.Ore.index,
+  'Range': material.Tree.index,
+  'Mage': material.Crystal.index
+}
+
+PROF_TO_ATK_TYPE = {
+  'Melee': 0,
+  'Range': 1,
+  'Mage': 2,
+}
+
+MAX_RESERVE_LEVEL = 6
+
+# CHECK ME: revisit the below constants
+def calc_weapon_price(level):
+  return min(99, np.random.randint(10) + int(level) * 25 - 10)
+
+def calc_armor_price(level):
+  return 3 + np.random.randint(3) + int(level - 1) * 4
+
+def calc_tool_price(level):
+  return int(level) * 6 + np.random.randint(3)
+
+def calc_ammo_price(level):
+  return max(1, int(level) - 1)
+
+# legal use/sell consumables-related
+# CHECK ME: how this goes along with _force_use/sell?
+N_USE = 2
```

Comment:
is this copied in action.py?

Reply:
I see N_USE is also in action.py. Haven't touched action.py, and thinking of merging it with `feature_extract.py` because action.py also needs many other helpers, which are already in FeatureExtractor.